ui/app/src/components/Navigation/PagesSearchAhead.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InputBase from '@material-ui/core/InputBase';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { useActiveSiteId, useContentTypeList } from '../../utils/hooks';
+import { search } from '../../services/search';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import { useAutocomplete } from '@material-ui/lab';
+import { SearchItem } from '../../models/Search';
+import clsx from 'clsx';
+import {
+  CircularProgress,
+  IconButton,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Paper
+} from '@material-ui/core';
+import LoadingState from '../SystemStatus/LoadingState';
+import EmptyState from '../SystemStatus/EmptyState';
+import Page from '../Icons/Page';
+import CloseIcon from '@material-ui/icons/Close';
+import { getPreviewURLFromPath } from '../../utils/path';
+import { FormattedMessage } from 'react-intl';
+import parse from 'autosuggest-highlight/parse';
+import match from 'autosuggest-highlight/match';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      width: '100%',
+      position: 'relative'
+    },
+    closeIcon: {
+      padding: '3px'
+    },
+    progress: {
+      position: 'absolute',
+      right: 0
+    },
+    inputRoot: {
+      width: '100%'
+    },
+    input: {},
+    paper: {
+      width: 400,
+      position: 'absolute',
+      right: '-52px',
+      top: '50px'
+    },
+    listBox: {
+      overflow: 'auto',
+      maxHeight: 600,
+      margin: 0,
+      padding: 0,
+      listStyle: 'none',
+      '& li[data-focus="true"]': {
+        backgroundColor: 'rgba(0, 0, 0, 0.04)'
+      },
+      '& li:active': {
+        backgroundColor: 'rgba(0, 0, 0, 0.04)',
+        color: 'white'
+      }
+    },
+    listItemIcon: {
+      minWidth: 'auto',
+      paddingRight: '16px'
+    },
+    EmptyImage: {
+      width: '100px'
+    },
+    highlighted: {
+      display: 'inline-block',
+      background: 'yellow'
+    }
+  })
+);
+
+export default function(props) {
+  const { value, placeholder, disabled, onEnter } = props;
+  const classes = useStyles({});
+  const onSearch$ = useMemo(() => new Subject<string>(), []);
```

Comment:
`useSubject<string>()`

Reply:
Its throws an error. React says do not call hooks inside useMemo